util/syspolicy/source/policy_store_windows_test.go

```diff
@@ -0,0 +1,394 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package source
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"golang.org/x/sys/windows"
+	"golang.org/x/sys/windows/registry"
+	"tailscale.com/util/cibuild"
+	"tailscale.com/util/mak"
+	"tailscale.com/util/syspolicy/setting"
+	"tailscale.com/util/winutil"
+	"tailscale.com/util/winutil/gp"
+)
+
+// subkeyStrings is a test type indicating that a string slice should be written
+// to the registry as multiple REG_SZ values under the setting's key,
+// rather than as a single REG_MULTI_SZ value under the group key.
+// This is the same format as ADMX use for string lists.
+type subkeyStrings []string
+
+type testPolicyValue struct {
+	name  setting.Key
+	value any
+}
+
+func TestLockUnlockPolicyStore(t *testing.T) {
```

Comment:
Might be good to make sure the test is verifying all the goroutines are gone at the end of the test and can't leak.

Reply:
Added
```go
// Make sure we don't leak goroutines
tstest.ResourceCheck(t)
```